app/views/admin/products/_form.html.haml

```diff
@@ -99,4 +99,6 @@
 
     .form_group#submit_div
       = f.submit class: 'btn btn-success'
-      = button_tag 'Cancel', type: :button, onclick: 'confirmProductCancel()', id: :cancel_button, class: 'btn btn-danger'
\ No newline at end of file
+      -if @product.persisted?
+        = link_to 'Delete', admin_product_path, {method: :delete, data: {confirm: 'Are you sure you want to delete this product?'}, class: 'btn btn-danger', id: :delete_button}
```

Comment:
I'm confused, I thought the browser doesn't understand methods other than post or get? How does it know what to do with delete?

Reply:
This is using the HTML DELETE request